src/posit/connect/users.py

```diff
@@ -57,6 +57,40 @@ def confirmed(self) -> bool:
     def locked(self) -> bool:
         return self.get("locked")  # type: ignore
 
+    def lock(self, *, force: bool = False):
+        """
+        Locks the user account.
+
+        .. warning:: You cannot unlock your own account. Once an account is locked, only an admin can unlock it.
+
+        Args:
+            force (bool, optional): If set to True, overrides lock protection allowing a user to lock their own account. Defaults to False.
+
+        Returns:
+            None
+        """
+        _me = me.get(self.config, self.session)
+        if _me.guid == self.guid and not force:
+            raise RuntimeError(
+                "You cannot lock your own account. Set force=True to override this behavior."
+            )
+        url = urls.append_path(self.config.url, f"v1/users/{self.guid}/lock")
+        body = {"locked": True}
+        self.session.post(url, json=body)
```

Comment:
Assuming this does not return the updated User record? That's too bad because `updated_time` should also be affected by this action. Not a huge deal but just wanted to note.

Reply:
Yes, the 200 response body is empty for `v1/users/{self.guid}/lock`. Adding a subsequent refresh here seems reasonable though.

```py
url = urls.append_path(self.config.url, f"v1/users/{self.guid}")
response = self.session.get(url)
self.update(**response.json())
```

Reply:
Related: https://github.com/posit-dev/posit-sdk-py/issues/141